src/engraving/libmscore/utils.cpp

```diff
@@ -1205,7 +1206,8 @@ bool moveDownWhenAddingStaves(EngravingItem* item, staff_idx_t startStaff, staff
     return true;
 }
 
-void collectChordsAndRest(Segment* segment, staff_idx_t staffIdx, std::vector<Chord*>& chords, std::vector<Rest*>& rests)
+void collectChordsAndRest(Segment* segment, staff_idx_t staffIdx, std::vector<Chord*>& chords, std::vector<Rest*>& rests,
```

Comment:
Now with checkChordsFromPreviousSegments this function looks weird: before this change, it behaved predictably and collected chords/rests from a given segment. But now it collects chords/rests in a very unpredictable range

Reply:
Uhm, good point. I'll move that part to a separate function

Reply:
done!